poprox-db/migrations/versions/2024_10_29_1406-585c9e8b4487_add_id_deleted_column_to_accounts.py

```diff
@@ -0,0 +1,26 @@
+"""add id_deleted column to accounts
+
+Revision ID: 585c9e8b4487
+Revises: f8667df4b049
+Create Date: 2024-10-29 14:06:53.219760
+
+"""
+
+from typing import Sequence, Union
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision: str = "585c9e8b4487"
+down_revision: Union[str, None] = "f8667df4b049"
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    op.add_column("accounts", sa.Column("is_deleted", sa.Boolean, nullable=True))
```

Comment:
Do we want nullable here? I recognize that you need to make it nullable to add it, but do we want the "final form" of this column to be nullable, or do we want it default-false? What's the difference between `is_deleted = null` and `is_deleted=False`?

(To do this if memory serves you do an add column that's nullable, update all rows to false, then modify to be non-null)

Reply:
I had the same approach as well, but went the other way. I shall make the changes

Reply:
That's not the wrong thing to do, but I suggested skipping it because of a defaults-related hassle:

SQLAlchemy treats non-nullable fields with database-level defaults as required, even though omitting them should theoretically be allowed. If this PR makes the new field non-nullable, it will also need to update the repository code to make sure that field is set whenever we store accounts.